### PR TITLE
Minor test fixes

### PR DESCRIPTION
**Pull request by glaslos**
Added some error handling to the test cases.
Updated the Go versions tested against in the travis config.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,7 +1,10 @@
...
-  - 1.4
-  - 1.5
-  - 1.6
-  - tip
+ - 1.5.4
+ - 1.6.4
+ - 1.7.6
+ - 1.8.3
+ - 1.9.0
+ - tip
```
Comment (dmitshur) [inline .travis.yml]: Suggestion, use this:

```YML
go:
  - 1.5.x
  - 1.6.x
  - 1.7.x
  - 1.8.x
  - 1.9.x
  - master
```

Reference: https://docs.travis-ci.com/user/languages/go/#Specifying-a-Go-version-to-use.

Comment (glaslos) [reply .travis.yml]: Thanks for the feedback, fixed!

File: .travis.yml [hunk 2]
```diff
@@ -1,7 +1,9 @@
...
-  - 1.4
-  - 1.5
-  - 1.6
-  - tip
+ - 1.5.x
+ - 1.6.x
+ - 1.7.x
+ - 1.8.x
+ - 1.9.x
+ - master
```
Comment (dmitshur) [inline .travis.yml]: Nitpick: YML files most commonly use 2 spaces for indentation, not one. Is there a specific reason you changed it from 2 to 1?

Comment (glaslos) [reply .travis.yml]: Fixed it and you were right to point it out. I was using then GitHub editor out of convenience which was set to 1 space for some reason and I didn't noticed in time.

**Comment by glaslos** [comment]
I adjusted the files as requested.

**Comment by glaslos** [comment]
Anything else blocking this PR?

**Comment by dmitshur** [comment]
I don't have any other suggestions for improvement. It looks like you need to get the project owner's attention. Ping @mattetti.

**Review by mattetti — APPROVED** [review]

**Comment by mattetti** [comment]
Thanks, sorry I missed the PR!

**Pull request merged**